File: clients/python/coyote/apis/rate_limit.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    RateLimitCheckIn,
    RateLimitCheckOut,
    RateLimitGetRemainingIn,
    RateLimitGetRemainingOut,
)
from .rate_limit_namespace import (
    RateLimitNamespace,
    RateLimitNamespaceAsync,
)


class RateLimitAsync(ApiBase):
    @property
    def namespace(self) -> RateLimitNamespaceAsync:
        return RateLimitNamespaceAsync(self._client)

    async def limit(
        self,
        rate_limit_check_in: RateLimitCheckIn,
    ) -> RateLimitCheckOut:
        """Rate Limiter Check and Consume"""
        body = rate_limit_check_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/rate-limit/limit",
            body=body,
            response_type=RateLimitCheckOut,
        )

    async def get_remaining(
        self,
        rate_limit_get_remaining_in: RateLimitGetRemainingIn,
    ) -> RateLimitGetRemainingOut:
        """Rate Limiter Get Remaining"""
        body = rate_limit_get_remaining_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1/rate-limit/get-remaining",
            body=body,
            response_type=RateLimitGetRemainingOut,
        )


class RateLimit(ApiBase):
    @property
    def namespace(self) -> RateLimitNamespace:
        return RateLimitNamespace(self._client)

    def limit(
        self,
        rate_limit_check_in: RateLimitCheckIn,
    ) -> RateLimitCheckOut:
        """Rate Limiter Check and Consume"""
        body = rate_limit_check_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1/rate-limit/limit",
            body=body,
            response_type=RateLimitCheckOut,
        )

    def get_remaining(
        self,
        rate_limit_get_remaining_in: RateLimitGetRemainingIn,
    ) -> RateLimitGetRemainingOut:
        """Rate Limiter Get Remaining"""
        body = rate_limit_get_remaining_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1/rate-limit/get-remaining",
            body=body,
            response_type=RateLimitGetRemainingOut,
        )

File: z-clients/java/src/main/java/com/svix/diom/apis/RateLimit.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.DiomException;
import com.svix.diom.HttpClient;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import com.svix.diom.models.RateLimitCheckIn;
import com.svix.diom.models.RateLimitCheckOut;
import com.svix.diom.models.RateLimitGetRemainingIn;
import com.svix.diom.models.RateLimitGetRemainingOut;
import com.svix.diom.models.RateLimitResetIn;
import com.svix.diom.models.RateLimitResetOut;

public class RateLimit {
    private final HttpClient client;

    public RateLimit(HttpClient client) {
        this.client = client;
    }

    public RateLimitNamespace namespace() {
        return new RateLimitNamespace(this.client);
    }

    /** Rate Limiter Check and Consume */
    public RateLimitCheckOut limit(
        final RateLimitCheckIn rateLimitCheckIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.rate-limit.limit",
            null,
            rateLimitCheckIn,
            RateLimitCheckOut.class
        );
    }

    /** Rate Limiter Get Remaining */
    public RateLimitGetRemainingOut getRemaining(
        final RateLimitGetRemainingIn rateLimitGetRemainingIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.rate-limit.get-remaining",
            null,
            rateLimitGetRemainingIn,
            RateLimitGetRemainingOut.class
        );
    }

    /** Rate Limiter Reset */
    public RateLimitResetOut reset(
        final RateLimitResetIn rateLimitResetIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.rate-limit.reset",
            null,
            rateLimitResetIn,
            RateLimitResetOut.class
        );
    }
}